src/strconv.c

```diff
@@ -0,0 +1,135 @@
+#include "../include/strconv.h"
+
+#include <string.h>
+#include <malloc.h>
+
+string_t *create_string(char *str, int max) {
+    string_t *nstr = NULL;
+    
+    if (str == NULL) {
+        nstr = malloc(sizeof(string_t));
+        
+        if (nstr == NULL) {
+            return NULL;
+        }
+
+        nstr->str = calloc(sizeof(char) * max, 1);
+
+        if (nstr->str == NULL) {
+            return NULL;
+        }
+
+        nstr->size = max;
+        return nstr;
+    }
+
+    int index = 0;
+
+    while (1) {
+        if (index == max) {
+            break;
+        }
+
+        if (str[index] == 0) {
+            break;
+        }
+
+        index++;
+    }
+
+    nstr = malloc(sizeof(string_t));
+    
+    if (nstr == NULL) {
+        return NULL;
+    }
+
+    nstr->str = malloc(sizeof(char) * index);
+
+    if (nstr->str == NULL) {
+        return NULL;
+    }
+
+    nstr->size = index;
+
+    for (int i = 0; i < index; i++) {
+        nstr->str[i] = str[i];
+    }
+
+    return nstr;
+}
+
+void free_string(string_t **str) {
+    free((*str)->str);
+    free((*str));
+
+    // To be safe.
+    *str = NULL;
+}
+
+void print_string(string_t *str, char *format) {
+    int fmtlen = strlen(format);
+
+    for (int i = 0; i < fmtlen; i++) {
+        if (format[i] == '%' && format[i+1] == 's') {
+            for (int i = 0; i < str->size; i++) {
+                printf("%c", str->str[i]);
+            }
+            // We increment i to get rid of s.
+            i++;
+        } else {
+            printf("%c", format[i]);
+        }
+    }
+}
+
+int check_len_by_8(string_t *str) {
```

Comment:
aici poate ar fi mai general daca am redenumi functia in "check_len_by_target" si sa dai target-ul ca parametru.

Reply:
@deian-balutoiu aici nu vad rostul sa avem un target variabil din moment ce totul trebuie sa fie divizibil cu 8. Cheia trebuie sa fie obligatoriu 8, dar pentru text trebuie sa aflam size-ul in legătură cu un multiplu de 8. Am putea sa adăugăm un target, dar mi se pare redundant.